src/Core/maps/src/Platform/iOS/MauiMKMapView.cs

```diff
@@ -0,0 +1,305 @@
+﻿using System;
+using System.Collections;
+using System.Linq;
+using CoreLocation;
+using MapKit;
+using Microsoft.Maui.Maps.Handlers;
+using Microsoft.Maui.Platform;
+using ObjCRuntime;
+using UIKit;
+
+namespace Microsoft.Maui.Maps.Platform
+{
+	internal class MauiMKMapView : MKMapView
+	{
+		IMapHandler? _handler;
+		object? _lastTouchedView;
+
+		public MauiMKMapView(IMapHandler handler)
+		{ 
+			_handler = handler;
```

Comment:
I wonder if this is the best way to intecract from the custom view backto the handler

Reply:
yea we kind of have a mixed way of interacting back with the handler. For example the layout stuff all uses `Func`'s to communicate so that they aren't tied to any handlers or `IView`

But we've done it this way a lot of places

The only thing I'd say here is to make sure to store the `_handler` into a WeakReference otherwise this will most likely cause a circular reference

Reply:
Done, use WeakReference